lib/streamlit/elements/image.py

```diff
@@ -288,22 +287,22 @@ def _ensure_image_size_and_format(
     MAXIMUM_CONTENT_WIDTH. Ensure the image's format corresponds to the given
     ImageFormat. Return the (possibly resized and reformatted) image bytes.
     """
-    image = Image.open(io.BytesIO(image_data))
-    actual_width, actual_height = image.size
+    pil_image = Image.open(io.BytesIO(image_data))
+    actual_width, actual_height = pil_image.size
 
     if width < 0 and actual_width > MAXIMUM_CONTENT_WIDTH:
         width = MAXIMUM_CONTENT_WIDTH
 
     if width > 0 and actual_width > width:
         # We need to resize the image.
         new_height = int(1.0 * actual_height * width / actual_width)
-        image = image.resize((width, new_height), resample=Image.BILINEAR)
-        return _PIL_to_bytes(image, format=image_format, quality=90)
+        pil_image = pil_image.resize((width, new_height), resample=Image.BILINEAR)
+        return _PIL_to_bytes(pil_image, format=image_format, quality=90)
 
-    ext = imghdr.what(None, image_data)
-    if ext != image_format.lower():
+    # pil_image.format should be defined since we call Image.open()
```

Comment:
Not sure what this comment is supposed to mean.

If it means that `pil_image.format` is not `None`, then that's false since if you pass it an image that's been created by or modified using the Pillow library, it'll be `None` since we have yet to save the image to a file. The format won't be defined until we call `_PIL_to_bytes` below, which will always be called because `pil_image.format` will be `None` but `image_format` will be a file extension.

If it means that you can always access `pil_image.format`, then I don't think it's a useful thing to mention since Pillow's API defines the `.format` attribute of an `Image` as `str | None`, meaning that it's always defined.

Reply:
Ah yeah my comment is not particularly clear. I meant the second one. I'll just remove it.